### PR TITLE
[master] Improved tsv parsing

### PR DESCRIPTION
**Pull request by mdouchement**
- Reduce by 14% memory usage
- Improved code structure
- Add example

### CODE REVIEW

File: examples/tsvsorter/main.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+	elapsed = time.Since(start)
+	fmt.Printf("Total time %s\n", elapsed)
+
+	return nil
+}
+
+func fail(err error) {
+	panic(err)
+}
```
Comment (dehau) [inline examples/tsvsorter/main.go]: why not using panic directly ?

File: tsv_parser.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package iosupport
+
+import (
+	"bytes"
+)
+
+// UnescapeSeparator cleans composed separator like \t
+func UnescapeSeparator(separator string) byte {
+	if separator == "\\t" {
```
Comment (dehau) [inline tsv_parser.go]: Why ?

File: tsv_parser.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+func TrimNewline(line []byte) []byte {
+	line = bytes.TrimRight(line, "\n")
+	line = bytes.TrimRight(line, "\r")
+	return bytes.TrimRight(line, "\n")
+}
+
+// A TsvParser reads records from a TSV-encoded file.
+type TsvParser struct {
+	*Scanner
+	err       error // Sticky error.
```
Comment (dehau) [inline tsv_parser.go]: sticky ?

File: tsv_parser.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	row       [][]byte
+}
+
+// NewTsvParser inatanciates a new TsvParser
+func NewTsvParser(sc *Scanner, separator byte) *TsvParser {
+	sc.Reset()
+	return &TsvParser{
+		Scanner:   sc,
+		Separator: []byte{separator},
+		QuoteChar: '"',
```
Comment (dehau) [inline tsv_parser.go]: if the `,` by design ?

File: tsv_parser.go (hunk 2 above)
Comment (mdouchement) [inline tsv_parser.go]: In some cases, `\t` is not rightly interpreted.

File: tsv_parser.go (hunk 4 above)
Comment (mdouchement) [inline tsv_parser.go]: Yep

File: tsv_parser.go (hunk 3 above)
Comment (mdouchement) [inline tsv_parser.go]: The error is not returned by the called method as usual.

**Pull request merged**